File: Type-3/JoseVitorSAmorim_4.py
def linhas():
    print("-" * len("Calculadora de IRPF 2026"))

def traco():
    print("~" * len("Calculadora de IRPF 2026"))

def IRPF2026(salario_bruto):
    linhas()
    desconto_simples = 607.20

    if salario_bruto <= 1500:
        inss = salario_bruto * 0.075
    elif salario_bruto <= 2800:
        inss = salario_bruto * 0.09
    elif salario_bruto <= 4000:
        inss = salario_bruto * 0.12
    else:
        inss = 929.59

    base_calculo = salario_bruto - desconto_simples

    if base_calculo <= 2428.80:
        aliquota = 0
        parcela_deduzir = 0
    elif base_calculo <= 2826.65:
        aliquota = 0.075
        parcela_deduzir = 182.16
    elif base_calculo <= 3751.05:
        aliquota = 0.15
        parcela_deduzir = 394.16
    elif base_calculo <= 4664.68:
        aliquota = 0.225
        parcela_deduzir = 675.49
    else:
        aliquota = 0.275
        parcela_deduzir = 988.73

    imposto_base = base_calculo * aliquota - parcela_deduzir

    if salario_bruto <= 5000:
        redutor_especial = imposto_base
    elif salario_bruto > 5000 and salario_bruto <= 7350:
        redutor_especial = 978.62 - (0.133145 * salario_bruto)
    else:
        redutor_especial = 0
    imposto_final = max(0, imposto_base - redutor_especial)
    salario_liquido = salario_bruto - inss - imposto_final
    return{
        "Salário Bruto": salario_bruto,
        "INSS": inss,
        "Base de Cálculo": base_calculo,
        "Alíquota": aliquota,
        "Parcela a Deduzir": parcela_deduzir,
        "Imposto Base": imposto_base,
        "Redutor Especial": redutor_especial,
        "Imposto Final": imposto_final,
        "Salário Líquido": salario_liquido
    }
traco()
print("Calculadora de IRPF 2026")
traco()
salario = float(input("Digite o salário bruto: "))
resultado = IRPF2026(salario)
for chave, valor in resultado.items():
    print(f"{chave}: R${valor:.2f}")
print("-" * len("Calculadora de IRPF 2026"))

File: Type-3/JoseVitorSAmorim_4.java
import java.util.Scanner;
import java.util.LinkedHashMap;
import java.util.Map;
public class JoseVitorSAmorim_4 {
    static final String TITULO = "Calculadora de IRPF 2026";

    static void linhas() {
        System.out.println("-".repeat(TITULO.length()));
    }

    static void traco() {
        System.out.println("~".repeat(TITULO.length()));
    }

    static Map<String, Double> IRPF2026(double salarioBruto) {
        linhas();
        double descontoSimples = 607.20;
        double inss = 0;
        if (salarioBruto <= 1500) {
            inss = salarioBruto * 0.075;
        } 
        else if (salarioBruto <= 2800) {
            inss = salarioBruto * 0.09;
        } 
        else if (salarioBruto <= 4000) {
            inss = salarioBruto * 0.12;
        } 
        else {
            inss = 929.59;
        }

        double baseCalculo = salarioBruto - descontoSimples;
        double aliquota = 0;
        double parcelaDeduzir = 0;

        if (baseCalculo <= 2428.80) {
            aliquota = 0;
            parcelaDeduzir = 0;
        } 
        else if (baseCalculo <= 2826.65) {
            aliquota = 0.075;
            parcelaDeduzir = 182.16;
        } 
        else if (baseCalculo <= 3751.05) {
            aliquota = 0.15;
            parcelaDeduzir = 394.16;
        } 
        else if (baseCalculo <= 4664.68) {
            aliquota = 0.225;
            parcelaDeduzir = 675.49;
        } 
        else {
            aliquota = 0.275;
            parcelaDeduzir = 988.73;
        }

        double impostoBase = (baseCalculo * aliquota) - parcelaDeduzir;
        double redutorEspecial = 0;

        if (salarioBruto <= 5000) {
            redutorEspecial = impostoBase;
        } 
        else if (salarioBruto > 5000 && salarioBruto <= 7350) {
            redutorEspecial = 978.62 - (0.133145 * salarioBruto);
        } 
        else {
            redutorEspecial = 0;
        }

        double impostoFinal = Math.max(0, impostoBase - redutorEspecial);
        double salarioLiquido = salarioBruto - inss - impostoFinal;

        Map<String, Double> resultado = new LinkedHashMap<>();
        resultado.put("Salário Bruto", salarioBruto);
        resultado.put("INSS", inss);
        resultado.put("Base de Cálculo", baseCalculo);
        resultado.put("Alíquota", aliquota);
        resultado.put("Parcela a Deduzir", parcelaDeduzir);
        resultado.put("Imposto Base", impostoBase);
        resultado.put("Redutor Especial", redutorEspecial);
        resultado.put("Imposto Final", impostoFinal);
        resultado.put("Salário Líquido", salarioLiquido);

        return resultado;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        traco();
        System.out.println(TITULO);
        traco();
        System.out.print("Digite o salário bruto: ");
        double salario = scanner.nextDouble();
        Map<String, Double> resultado = IRPF2026(salario);
        for (Map.Entry<String, Double> item : resultado.entrySet()) {
            System.out.printf("%s: R$%.2f\n", item.getKey(), item.getValue());
        }
        linhas();
        scanner.close();
    }
}
